### ISSUE TITLE
autoware.universe and Robosense Lidar

### ISSUE THREAD

**Issue by mosss-code** [TZECHIN6/robosense_description#1]
博主你好，关于autoware.universe与robosense Lidar适配的问题你解决了吗？
Hello, have you resolved the issue of compatibility between autoware.universe and robosense Lidar?

**Comment by TZECHIN6** [comment]
So far, yes but not ideal. I am still waiting for the official release, as they should have a better understanding to their product. This description is highly reference to the velodyne's one.

**Comment by mosss-code** [comment]
感谢回答，目前我在尝试Robosense RS16与autoware.universe集成，但是失败了。我尝试使用ros_rslidar发送原始数据给velodyne_pointcloud，但是处理后的结果依然无法使用。
Thanks for the answer, currently I'm trying to integrate Robosense RS16 with autoware.universe, but failed. I try to use ros_rslidar to send raw data to velodyne_pointcloud, but the processed result is still unusable.

**Comment by TZECHIN6** [comment]
If u can launch rs LiDAR with autoware I believe the description file is working at least. First thing to check is to echo the original /rslidar_points to see is there scan data, if yes, then this might caused by the missing data field of rslidar driver. So when you echo the final output of precoessing node, it should be empty.

**Comment by DiTianba** [comment]
I used the official driver [rslidar_sdk](https://github.com/RoboSense-LiDAR/rslidar_sdk.git) provided.I added the CAD field for the topic /rslidar_points it eventually outputs and passed the input topic directly into this cropbox component.
```
 nodes.append(
    ComposableNode(
        package="pointcloud_preprocessor",
        plugin="pointcloud_preprocessor::CropBoxFilterComponent",
        name="crop_box_filter_self",
        remappings=[
            ("input", "/rslidar_points),
            ("output", "self_cropped/pointcloud_ex"),
        ],
        parameters=[cropbox_parameters],
        extra_arguments=[{"use_intra_process_comms": LaunchConfiguration("use_intra_process")}],
    )
)
```
The self_cropped/pointcloud_ex topic has data output. But I haven't done a real car test yet.

**Comment by TZECHIN6** [comment]
@DiTianba yes, you have to add extra data field into the original driver before passing to the processing node. Btw, back to the description file, have you created your own? Would u mind sharing to the community? Thanks

**Comment by DiTianba** [comment]
@TZECHIN6 I just changed the topic /rslidar_points output by rslidar_sdk to /sensing/lidar/top/rslidar_points, and commented out the code about velodyne.

**Comment by TZECHIN6** [comment]
oh, so you havent give a TF to the base_link of your robosense lidar? I think its quite important for your robot to understand the reference of each lidar points, otherwise the localization and navigation would have problems i believe.
